Form/DoctrineMongoDBTypeGuesser.php

```diff
@@ -39,9 +40,12 @@ class DoctrineMongoDBTypeGuesser implements FormTypeGuesserInterface
 
     private $cache = array();
 
+    private $typeFQCN;
+
     public function __construct(ManagerRegistry $registry)
     {
         $this->registry = $registry;
+        $this->typeFQCN = version_compare(Kernel::VERSION, '2.8.0', '>=');
```

Comment:
this should not detect the kernel version but the version of the Form component. It is perfectly valid to install the Form 2.8 component with HttpKernel 2.7.

Use ``method_exists('\Symfony\Component\Form\AbstractType', 'getBlockPrefix')`` to detect the new form component